pyfstat/core.py

```diff
@@ -1676,9 +1695,9 @@ def _get_per_segment_det_stat(self, tstart):
                 2 * FS.F_mn.data[0][0], self.twoFX, self.BSGLSetup
             )
             d_detStat = log10_BSGL / np.log10(np.exp(1))
-        if np.isnan(d_detStat):
+        if np.isnan(np.sum(d_detStat)):
             logging.debug("NaNs in semi-coherent twoF treated as zero")
```

Comment:
Also unrelated to your change, but a quick fix: this should actually say `in semi-coherent detection statistic`.